test/helpers/scene-builder.ts

```diff
@@ -228,4 +243,44 @@ export class SceneBuilder {
 
     return nodes;
   }
+
+  /**
+   * The functions below are primarily used for testing dual output
+   */
+
+  confirmDualOutputCollection(): boolean {
+    const nodes = this.getSceneSchema();
+
+    // if scene node map doesn't exist, return
+    const activeSceneId = this.scenesService.activeSceneId;
+    const sceneNodeMaps = this.scenesService?.sceneNodeMaps;
+    if (!sceneNodeMaps) return false;
+
+    const nodeMap = sceneNodeMaps[activeSceneId];
+    if (!nodeMap) return false;
+
+    // confirm all horizontal nodes have a partner vertical node
+    return nodes.reduce((hasPartner: boolean, node: ISceneBuilderNode) => {
+      if (node?.display === 'horizontal' && !nodeMap[node.id]) {
+        hasPartner = false;
+      }
+      return hasPartner;
+    }, true);
```

Comment:
this is pretty nitpicky and doesn't require a refactor but just letting you know! JS has `array.every` and `array.some` methods now that seem ideal for a lot of these places you're using reduce